pkg/controller/certificates/groupapprove_test.go

```diff
@@ -69,3 +75,140 @@ func TestHasKubeletUsages(t *testing.T) {
 		}
 	}
 }
+
+func TestAutoApprove(t *testing.T) {
+	certificateRequestPEM, err := generateCertificateRequest("system:node:test-node-name", []string{"system:nodes"})
+	if err != nil {
+		t.Fatalf("Unable to generate a certificate request: %v", err)
+	}
+
+	workingCertificateSigningRequest := certificates.CertificateSigningRequest{
+		Spec: certificates.CertificateSigningRequestSpec{
+			Groups:  []string{"approved-group"},
+			Request: certificateRequestPEM,
+			Usages: []certificates.KeyUsage{
+				certificates.UsageKeyEncipherment,
+				certificates.UsageDigitalSignature,
+				certificates.UsageClientAuth,
+			},
+		},
+		Status: certificates.CertificateSigningRequestStatus{
+			Conditions: []certificates.CertificateSigningRequestCondition{},
+		},
+	}
+	testCases := []struct {
+		description                string
+		approvalGroup              string
+		csr                        certificates.CertificateSigningRequest
+		expectError                bool
+		expectedApproved           bool
+		expectedDenied             bool
+		expectedNumberOfConditions int
+	}{{
+		description:                "Matching approval group",
+		approvalGroup:              "approved-group",
+		csr:                        workingCertificateSigningRequest,
+		expectError:                false,
+		expectedApproved:           true,
+		expectedDenied:             false,
+		expectedNumberOfConditions: 1,
+	}}
+	successTestCase := testCases[0]
+
+	tc := successTestCase
+	tc.description = "No approval group configured"
```

Comment:
Why construct the list of tests this way instead of during intialization like other tests?